tests/e2e/vmc_scale_gc_workers.go

```diff
@@ -0,0 +1,324 @@
+		vmcUser           string
+	)
+
+	ginkgo.BeforeEach(func() {
+		client = f.ClientSet
+		namespace = getNamespaceToRunTests(f)
+		svcClient, svNamespace = getSvcClientAndNamespace()
+		nodeList, err := fnodes.GetReadySchedulableNodes(f.ClientSet)
+		framework.ExpectNoError(err, "Unable to find ready and schedulable Node")
+		storagePolicyName = GetAndExpectStringEnvVar(envStoragePolicyNameForSharedDatastores)
+		if !(len(nodeList.Items) > 0) {
+			framework.Failf("Unable to find ready and schedulable Node")
+		}
+		bootstrap()
+		setResourceQuota(svcClient, svNamespace, rqLimit)
+		scParameters = make(map[string]string)
+		storagePolicyName = GetAndExpectStringEnvVar(envStoragePolicyNameForSharedDatastores)
+	})
+	ginkgo.AfterEach(func() {
+		svcClient, svNamespace := getSvcClientAndNamespace()
+		setResourceQuota(svcClient, svNamespace, defaultrqLimit)
+	})
+
+	/*
+		Test to Scale TKG worker nodes
+		Steps
+			1. Create 3 Statefulset pods, PVC with the above created SC and validate PVC is in bound phase
+			2. Scale up the worker nodes to 5
+			3. Scale up the Statefulset pods to 8
+			4. Verify volumes are attached to the nodes
+			5. Wait for some time to scale up operations to complete
+			6. Verify PVC and POD should be in bound and running state
+			7. Scale down the worker nodes to 2
+			8. Wait for some time to scale down operations to complete
+			9. Verify POD should be in running state
+			10. Verify volumes are attached to the nodes
+			11. Scale down the sts pods to 0
+			12. Verify volumes are detached from the nodes
+			10. Delete the statefulsets
+	*/
+
+	ginkgo.It("[vmc] Scale TKG Worker nodes", func() {
+		var sc *storagev1.StorageClass
+		var err error
+		var wcpToken string
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		vmcUser = os.Getenv("VMC_USER")
+		if vmcUser == "" {
+			ginkgo.Skip("Env vmcUser is missing")
+		}
+
+		ginkgo.By("Creating StorageClass for Statefulset")
+		scParameters[svStorageClassName] = storagePolicyName
+		scSpec := getVSphereStorageClassSpec(defaultNginxStorageClassName, scParameters, nil, "", "", false)
+		sc, err = client.StorageV1().StorageClasses().Create(ctx, scSpec, metav1.CreateOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		defer func() {
+			err := client.StorageV1().StorageClasses().Delete(ctx, sc.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Creating statefulset")
+		statefulset := fss.CreateStatefulSet(client, manifestPath, namespace)
+		defer func() {
+			ginkgo.By(fmt.Sprintf("Deleting all statefulsets in namespace: %v", namespace))
+			fss.DeleteAllStatefulSets(client, namespace)
+			if supervisorCluster {
+				ginkgo.By(fmt.Sprintf("Deleting service nginx in namespace: %v", namespace))
+				err := client.CoreV1().Services(namespace).Delete(ctx, servicename, *metav1.NewDeleteOptions(0))
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			}
```

Comment:
where are we creating the service?

Reply:
In `CreateStatefulSet` method service is being created